earthmover/nodes/source.py

```diff
@@ -266,13 +265,29 @@ def __get_skiprows(config: 'YamlMapping'):
             _header_rows = config.get('header_rows', 1)
             return int(_header_rows) - 1  # If header_rows = 1, skip none.
 
+        def __read_fwf(file: str, config: 'YamlMapping'):
+            colspec_file = config.get('colspec_file')
+            if colspec_file:
+                try:
+                    file_format = pd.read_csv(os.path.join(os.path.dirname(self.config.__file__), colspec_file))
```

Comment:
Is this approach for ascertaining filepath directories consistent with the rest of the project? If so, maybe we should move this logic to a helper.

Reply:
I'm not sure I can say whether it's consistent as such; it's needed in order to properly find the colspec file when using project composition. We do use the same construction once elsewhere but that's in a separate class. I've added a comment explaining this but I think it's too rare a usage to justify a separate function as of now